src/core_modules/capture-core/components/Pages/Enrollment/EnrollmentPage.epics.js

```diff
@@ -39,172 +59,287 @@ const enrollmentsQuery = (teiId, programId) => ({
+    return teiId && fetchStatus.teiId === selectionStatus.READY;
+};
+
+const programIdReady = (store: ReduxStore): boolean => {
+    const { fetchStatus } = store.value.enrollmentPage;
+    return fetchStatus.programId === selectionStatus.READY;
+};
+
+const enrollmentIdLoaded = (enrollmentId: string, enrollments: ?Array<Object>) =>
+    enrollments && enrollments.some(enrollment => enrollment.enrollment === enrollmentId);
+
+
+// The verification epics which are triggered by the completion of
+// async requests (e.g. verifyEnrollmentIdSuccessEpic) are not subject
+// to race conditions with other async requests because the chain of
+// epics and reducers triggered by the completion is resolved
+// synchronously before moving on to the next completed request.
+
+// Note that the reason for doing the verification in a separate epic
+// is to make sure we are using the most recent version of the redux
+// store (but we might in fact have access to the most recent state
+// using the old store object as well).
+
+// Epics for enrollmentId
+export const changedEnrollmentIdEpic = (action$: InputObservable, store: ReduxStore) =>
+    action$.pipe(
+        ofType(enrollmentPageActionTypes.PROCESS_ENROLLMENT_ID),
+        filter(({ payload: enrollmentId }) =>
+            ((!enrollmentId || enrollmentId === 'AUTO') ?
+                store.value.enrollmentPage.enrollmentId :
+                !enrollmentIdLoaded(enrollmentId, store.value.enrollmentPage.enrollments))),
+        map(({ payload: enrollmentId }) => {
+            if (!enrollmentId || enrollmentId === 'AUTO') {
+                const { programId, teiId } = getLocationQuery();
+                return resetEnrollmentId({ programId, teiId });
+            }
+            return fetchEnrollmentId(enrollmentId);
+        }),
+    );
+
+export const fetchEnrollmentIdEpic = (action$: InputObservable, store: ReduxStore, { querySingleResource }: ApiUtils) =>
+    action$.pipe(
+        ofType(enrollmentPageActionTypes.FETCH_ENROLLMENT_ID),
+        concatMap(({ payload: { enrollmentId } }) =>
+            querySingleResource(enrollmentIdQuery(enrollmentId))
+                .then(result => verifyEnrollmentIdSuccess({ enrollmentId, ...result }))
+                .catch(() => fetchEnrollmentIdError(enrollmentId))),
+    );
+
+export const verifyEnrollmentIdSuccessEpic = (action$: InputObservable, store: ReduxStore) =>
+    action$.pipe(
+        ofType(enrollmentPageActionTypes.VERIFY_ENROLLMENT_ID_SUCCESS),
+        filter(({ payload }) => payload.enrollmentId === store.value.enrollmentPage.enrollmentId),
+        map(({ payload }) => fetchEnrollmentIdSuccess(payload)),
+    );
+
+export const enrollmentIdErrorEpic = (action$: InputObservable) =>
+    action$.pipe(
+        ofType(enrollmentPageActionTypes.FETCH_ENROLLMENT_ID_ERROR),
+        map(({ payload: { enrollmentId } }) =>
+            showErrorViewOnEnrollmentPage({
+                error: i18n.t('Invalid enrollment id {{enrollmentId}}.', {
```

Comment:
Hi, can we use the same wording as in `fetchTeiErrorEpic` and `programIdErrorEpic` (e.g. `i18n.t('Enrollment with id "{{enrollmentId}}" does not exist', {enrollmentId })`?

I also noticed that when `orgUnitId` does not exist, the red error message is not displayed and a loading spinner is displayed instead. Is it possible to employ the same error message pattern used by the other URL parameters?

<img width="1190" alt="Screenshot 2024-02-09 at 14 26 26" src="https://github.com/dhis2/capture-app/assets/6817073/6bfcd474-b709-455a-b152-5432ad9be185">

Thanks!

Reply:
The handling of invalid orgUnitId implemented below doesn't look good enough on review; the scope selector shows the last selected orgunit, so to avoid confusion the error message should somehow indicate that it's the value in the url that is the problem.
Suggestion: `i18n.t('Organisation unit with id "{{orgUnitId}}" could not be retrieved', { orgUnitId })`.

Reply:
I agree that this is a bit confusing. Can you investigate if is possible to remove the last selected org unit from the scope selector?

- home page invalid `orgUnitId` handling

<img width="1017" alt="Screenshot 2024-02-15 at 09 30 19" src="https://github.com/dhis2/capture-app/assets/6817073/cf5f5de5-e38e-40dd-9ac6-a9e8f209c96d">


- new page invalid `orgUnitId` handling. The ScopeSelector is not displayed at all on this page. Perhaps this could also be an option for the enrollment page too?


<img width="779" alt="Screenshot 2024-02-15 at 09 31 45" src="https://github.com/dhis2/capture-app/assets/6817073/7706cc6f-240b-426a-a930-a0949971d120">

WDYT? Thanks!

Reply:
> Can you investigate if is possible to remove the last selected org unit from the scope selector?

I went with this solution.